### PR TITLE
Prototype

### PR DESCRIPTION
**Pull request by blogh**
An attempt at managing bytes decoding in pg_activity to hopefully stop the slew of encoding report on this topic.

### CODE REVIEW

**Comment by blogh** [comment]
@ denis is the approch sound at all ?

File: pgactivity/psycopg_utils.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            x = self._index[x]
+        val = super().__getitem__(x)
+        if isinstance(val, memoryview):
+            val = bytes(val)
+        if isinstance(val, bytes):
+            if self.__contains__("encoding"):
+                enc = super().__getitem__(self._index["encoding"])
+                enc = enc.decode("utf-8", "backslashreplace")
+            else:
+                enc = "utf-8"
```
[the remark this answers is not in the document]
Comment (blogh) [reply pgactivity/psycopg_utils.py:53]: guc `server_encoding`

File: pgactivity/types.py [hunk 2]
```diff
@@ -909,6 +909,7 @@ class BaseProcess:
     query: Optional[str]
```
[the remark this answers is not in the document]
Comment (blogh) [reply pgactivity/types.py:909]: The problem is that all strings become bytes so we need to convert them before using them.

[the remark this answers is not in the document]
Comment (dlax) [reply pgactivity/types.py:909]: Not *all strings*, just the one from the `query` column so this is fine I think.

Also, as mentioned offline, the previous approach relying on `convert_from(..., 'UTF-8')` assumes utf-8 and possibly this thus breaks if we don't have utf-8 when displaying in user's environment.

**Pull request closed without merging**